datatransfer/impl/graphsync/graphsync.go

```diff
@@ -109,24 +109,28 @@ func (impl *graphsyncImpl) OpenPushDataChannel(ctx context.Context, to peer.ID,
 	if err != nil {
 		return datatransfer.ChannelID{}, err
 	}
-	chid := impl.createNewChannel(tid, to, baseCid, selector, voucher)
+	chid := impl.createNewChannel(tid, baseCid, selector, voucher, to, "", to)
```

Comment:
So this is my bad -- the variable "To" in a channel ID is a bad name and needs to change. I'm sorry about that. It should be called "Initiator", and whether it's push or pull, the the side one which the request starts. (i.e. who calls OpenPushDataChannel or OpenPullDataChannel). So, in the two cases here, "initiator" should be the ID of the node itself -- "" as you put it here. (btw, you can get the peer.ID from the libp2p host passed into the constructor -- by calling host.ID())

When a request is received, you should save a channel state as well -- but this time make the initiator the other peer.

The whole idea here is for Initiator + TransferID to make a unique pair. Which means also you should change generateTransferID to generate sequentially.

Reply:
I don't know if you want to do this all now or later, but it all needs to happen to make all the tests pass I think.

Reply:
I'll make a separate, cleanup PR to address separating out receiver and fixing naming + transfer ID generation. I think it's a good one to take care of all the cans we've been kicking down the road